UnityGroceries-Real Documentation (#23)
Unity-Technologies/SynthDet: docs/UnityGroceriesRealWorld.md

# UnityGroceries-RealWorld Dataset

As part of this project, we collected a real world dataset containing 1267 images of the 63 target grocery items. This dataset is primarily used to assess the real-world performance of models trained on synthetic data with a model trained on real data. This dataset can be downloaded [here](https://storage.googleapis.com/datasetinsights/data/groceries/v3.zip).

## Dataset Collection

> To take the photos for the dataset, we devised a strategy to vary object selection, placement, lighting, and background. We found that we, the humans, needed more humans and a better strategy to randomize more thoroughly and faster. This new strategy is what is outlined.

Reviewer: > We found that we, the humans, needed more humans
lol! This was a joke actually. It is not a serious-sounding sentence :) how about "We iterated on our strategy a handful of times throughout the process to maximize our randomness and speed of data collection. The final strategy which we used for most of the images is described below."

Author: That's much better!

# UnityGroceries-RealWorld Dataset

> As part of this project, we collected a real world dataset containing 1267 images of the 63 corresponding grocery items. This dataset is primary used to compare performance of model trained on the Synthetic counterparts. This dataset can be downloaded [here](https://storage.googleapis.com/datasetinsights/data/groceries/v3.zip).

Reviewer: ```suggestion
As part of this project, we collected a real world dataset containing 1267 images of the 63 target grocery items. This dataset is primarily used to assess the real-world performance of models trained on synthetic data with a model trained on real data. This dataset can be downloaded [here](https://storage.googleapis.com/datasetinsights/data/groceries/v3.zip).
```

> ## dataset collection

Reviewer: ```suggestion
## Dataset Collection
```

> To take the photos for the UnityGroceries-RealWorld dataset, we devised a strategy to vary object selection, placement, lighting, and background. After a few test shoots, we realized that major barriers to randomness of placement and object selection were the actual humans doing the selecting and placing. Hence, planning on how to randomize while taking photos became highly important, and something we did before taking photos.

Reviewer: If you want to use the proper name for the dataset, introduce it at the beginning of the article. I think it sounds fine without "UnityGroceries-RealWorld".

Author: Agree. I removed the dataset name here. We don't need to repeat the name here.

> To take the photos for the UnityGroceries-RealWorld dataset, we devised a strategy to vary object selection, placement, lighting, and background. After a few test shoots, we realized that major barriers to randomness of placement and object selection were the actual humans doing the selecting and placing. Hence, planning on how to randomize while taking photos became highly important, and something we did before taking photos.

Reviewer: >After a few test shoots, we realized that major barriers to randomness of placement and object selection were the actual humans doing the selecting and placing. Hence, planning on how to randomize while taking photos became highly important, and something we did before taking photos.

I don't understand this part. What was the issue you discovered? If you want to include this section, be more specific about the previous and new method.

Author: @SCRktkt I am also a little confused about this part. It seems this sentence can be safely removed, as the "placement strategy" have be described in detail below.

Reviewer 2: We found that we, the humans, needed more humans and a better strategy to randomize more thoroughly and faster. This new strategy is what is outlined.

> For our photography sessions, we gathered all sixty three grocery objects, and staked out five locations to take the photos. These were:

Reviewer: ```suggestion
To provide a mix of lighting and backgrounds, We chose five locations with to take the photos. These were:
```
"Staked out" doesn't quite convey the action here. Here's a suggestion.

- A south facing large space with tall windows. This gave us strong daylight across the objects, and deep shadows.
- A west facing space with smaller windows, and an arrangement of furniture. This gave us bounced light and varied background colors and contours.
- A windowless space with variable lighting. This allowed us to create challenging one-sided and colored lighting.
- A second south facing space with partitions blocking the direct light. This gave us a blending of artificial and natural light, plus additional background objects.

> - A north facing alcove, with white laminate cabinets and countertops. This gave us strong artificial lighting and featureless areas to place objects.

Reviewer: Good! Nice and specific.

> To make sure we were randomizing placement and combinations of objects for the photos, we gathered several willing volunteers to assist, with one person as the photographer. In each of the locations described above, we used the following strategy:

Reviewer: ```suggestion
To make sure we had a variety of placement strategies and combinations of objects for the photos, we gathered several willing volunteers to assist. One person acted as the photographer. In each of the locations described above, we used the following strategy:
```

- Each volunteer started with eight objects, and placed them in different locales.
- Some objects were placed nicely, some were dropped or thrown lightly.
- The photographer would step around the space and take a picture of each volunteer’s arrangement, varying the angle each time.
- After the photo was taken, each volunteer would mix up the objects, hand placing some and dropping or lightly throwing others to create new arrangements.
- After the cycle was repeated eight times, each volunteer would hand an object to another volunteer, and receive one from a different volunteer.
- Once eight full cycles of object swapping and randomizing were completed, we gathered the groceries and moved to the next location to repeat the process.

> To take the pictures for the UnityGroceries-RealWorld dataset, we used a Canon T5i camera with a standard 35mm lens and no filters. The images were shot at the native resolution of 5184x3456 pixels, and saved as JPEG files at maximum quality. For all of the shots, White Balance, ISO, aperture, and shutter speed were set to Automatic, and no flash was used. All of the shots were handheld, with the photographer bracing on furniture or floor for stability wherever possible.

Reviewer: ```suggestion
For shooting, we used a Canon T5i camera with a standard 35mm lens and no filters. The images were shot at the native resolution of 5184x3456 pixels, and saved as JPEG files at maximum quality. For all of the shots, White Balance, ISO, aperture, and shutter speed were set to Automatic, and no flash was used. All of the shots were handheld, with the photographer bracing on furniture or floor for stability wherever possible.
```

> ## bounding box annotations

Reviewer: ```suggestion
## Bounding Box Annotations
```

We followed similar guidelines from [VOC](http://host.robots.ox.ac.uk/pascal/VOC/voc2011/guidelines.html) dataset. In addition to VOC guidelines, annotators are also required to follow these guidelines:

- If parts of the object are occluded, the bounding box should not extend to the occluded parts. We only draw bounding boxes that enclose visible parts of the object.
- If the object is split into 2 segments due occlusion, one should draw a bounding box that include both segments.
- If the approximate visibility of a given object is less than 10% due to edge cropping or occlusion from other objects, this object is excluded from annotations.
- If more than 30% of the objects in the image are blurred due to motion blur or camera focus, we regard this image as “unidentifiable”. These images are removed from the dataset.

> ## dataset splits

Reviewer: ```suggestion
## Dataset Splits
```
